### PR TITLE
refactor(completion): extract strategies from CompletionProvider

### LINKED ISSUE #952: refactor(groovy-lsp): Extract ImportCompletionStrategy from CompletionProvider
## Summary

Extract import completion logic from `CompletionProvider` into a dedicated `ImportCompletionStrategy` class.

## Problem

`CompletionProvider` is 1,209 lines, exceeding the 600-line limit by 201%. The import completion logic (~360 lines) handles:
- Regular import completions
- Static import completions
- Static member completions (methods and fields)
- Workspace and classpath class lookups

## Proposed Solution

1. Create `ImportCompletionStrategy` implementing `CompletionStrategy` interface
2. Move methods:
   - `addImportCompletions()`
   - `tryAddStaticMemberCompletions()`
   - `addWorkspaceStaticMemberCompletions()`
   - `addStaticMethodCompletions()` (both overloads)
   - `addStaticFieldCompletions()` (both overloads)
   - `addStaticPropertyCompletions()`
3. Move `ImportCompletionContext` data class
4. Update `CompletionProvider.buildCompletionsList()` to use strategy

## Test Refactoring Required

- [ ] Create `ImportCompletionStrategyTest.kt` with tests for:
  - Regular import prefix matching
  - Static import class member lookup
  - Static method completions with prefix filtering
  - Static field/constant completions
  - Workspace class static member lookup
  - Empty/blank prefix handling
- [ ] Move import-specific tests from `CompletionProviderTest`
- [ ] Add edge case tests (non-existent class, no static members)

## Acceptance Criteria

- [ ] `ImportCompletionStrategy` implements `CompletionStrategy`
- [ ] All import completion features work unchanged
- [ ] Unit test coverage ≥70%
- [ ] `CompletionProvider` reduced by ~360 lines
- [ ] Existing completion tests pass

## References

Part of god class refactoring effort for `CompletionProvider` (1,209 → ~400 lines target)

### LINKED ISSUE #953: refactor(groovy-lsp): Extract MemberAccessCompletionStrategy from CompletionProvider
## Summary

Extract member access completion logic from `CompletionProvider` into a dedicated `MemberAccessCompletionStrategy` class.

## Problem

`CompletionProvider` contains ~200 lines of member access handling that coordinates:
- Map literal key completions
- Jenkins global variable properties
- Workspace member lookups (cross-file)
- GDK method additions
- Classpath method lookups

## Proposed Solution

1. Create `MemberAccessCompletionStrategy` implementing `CompletionStrategy`
2. Move methods:
   - `handleMemberAccessContext()`
   - `addMapLiteralKeyCompletions()`
   - `addJenkinsGlobalVariablePropertyCompletions()`
   - `addWorkspaceMembers()`
   - `addGdkMethods()` / `addGdkMethodsForSingleType()`
   - `addClasspathMethods()` / `addClasspathMethodsForSingleType()`
   - `resolveWorkspaceClassFqns()`
   - `findWorkspaceClassFqnsBySimpleName()`
3. Update `CompletionProvider` to delegate `ContextType.MemberAccess` handling

## Test Refactoring Required

- [ ] Create `MemberAccessCompletionStrategyTest.kt` with tests for:
  - GDK method completions (e.g., `String.each()`)
  - Classpath method completions (e.g., `String.substring()`)
  - Workspace cross-file member completions
  - Map literal key completions
  - Jenkins global variable property completions
  - Type resolution from imports/package
- [ ] Move member access tests from `CompletionProviderTest`
- [ ] Add integration test for cross-file completion

## Acceptance Criteria

- [ ] `MemberAccessCompletionStrategy` is standalone
- [ ] GDK + classpath + workspace members work
- [ ] Unit test coverage ≥70%
- [ ] `CompletionProvider` reduced by ~200 lines
- [ ] Existing member access tests pass

## References

Part of god class refactoring effort for `CompletionProvider`

### LINKED ISSUE #954: refactor(groovy-lsp): Extract SpockCompletionStrategy from CompletionProvider
## Summary

Extract Spock test framework completion logic from `CompletionProvider` into a dedicated `SpockCompletionStrategy` class.

## Problem

`CompletionProvider` contains Spock-specific block label completion logic (~80 lines) that:
- Detects Spock spec context
- Provides block labels (given:, when:, then:, expect:, etc.)
- Uses heuristics for cursor position detection

## Proposed Solution

1. Create `SpockCompletionStrategy` implementing `CompletionStrategy`
2. Move methods:
   - `addSpockBlockLabelsIfApplicable()`
   - `isLineIndentOnlyBeforeCursor()`
   - `offsetAt()`
3. Move Spock block label definitions
4. Integrate with existing strategy aggregation in `buildCompletionsList()`

## Test Refactoring Required

- [ ] Create `SpockCompletionStrategyTest.kt` with tests for:
  - Block label completions in Spock spec
  - No completions in non-Spock files
  - No completions when cursor is mid-expression
  - No completions inside strings/comments
  - Correct sort order (ahead of keywords)
- [ ] Move Spock-specific tests from `CompletionProviderTest`
- [ ] Add edge cases (nested specs, trait specs)

## Acceptance Criteria

- [ ] `SpockCompletionStrategy` implements `CompletionStrategy`
- [ ] Spock block labels appear correctly in specs
- [ ] Unit test coverage ≥70%
- [ ] `CompletionProvider` reduced by ~80 lines
- [ ] Spock completion tests pass

## References

Part of god class refactoring effort for `CompletionProvider`

### PR DESCRIPTION
**Pull request by albertocavalcante**
## Summary

This PR extracts three completion strategies from `CompletionProvider` into dedicated, testable classes as part of Phase 2 of the god class refactoring effort:

- **ImportCompletionStrategy** (381 lines): Handles import statement completions, including automatic FQN-to-import suggestions
- **MemberAccessCompletionStrategy** (461 lines): Handles member and field access completions, including method calls and property access
- **SpockCompletionStrategy** (128 lines): Handles Spock testing framework-specific completions (blocks, methods, annotations)

**Impact:**
- Reduces `CompletionProvider` from 1,209 lines to 602 lines (~50% reduction)
- Improves testability with focused unit tests for each strategy (1,161 lines of new test coverage)
- Maintains backward compatibility—no behavior changes

**Related Issues:**
- Closes #952 (ImportCompletionStrategy)
- Closes #953 (MemberAccessCompletionStrategy)
- Closes #954 (SpockCompletionStrategy)

## Test Plan

- [x] All existing tests pass
- [x] New unit tests added for each strategy:
  - `ImportCompletionStrategyTest` (270 lines): Tests import completions, FQN suggestions, deduplication
  - `MemberAccessCompletionStrategyTest` (501 lines): Tests method/field access, qualified names, member resolution
  - `SpockCompletionStrategyTest` (390 lines): Tests Spock blocks, methods, annotations
- [x] Pre-commit hooks pass (spotless, lint)
- [x] Manual verification: Completion features work as expected in IDE

### CODE REVIEW

File: groovy-lsp/src/main/kotlin/com/github/albertocavalcante/groovylsp/providers/completion/CompletionProvider.kt [hunk 1]
```diff
@@ -308,10 +270,31 @@ object CompletionProvider {
...
+                SpockCompletionStrategy(),
+                GroovyCompletionStrategy(),
+            )
+            GroovyMode.AUTO -> listOf(
+                memberAccessStrategy,
+                JenkinsCompletionStrategy(),
+                SpockCompletionStrategy(),
+                GroovyCompletionStrategy(),
+            )
+        }
```
[the remark this answers is not in the document]
Comment (albertocavalcante) [reply groovy-lsp/src/main/kotlin/com/github/albertocavalcante/groovylsp/providers/completion/CompletionProvider.kt]: I prefer the current one, declarative and easier to read despite the duplication
